From b98bcd46b3678fef89c989312d25154bbfcc4a26 Mon Sep 17 00:00:00 2001
From: Gerald Combs <gerald@wireshark.org>
Date: Fri, 2 Jan 2015 14:06:42 -0800
Subject: [PATCH 1/2] Make sure we don't underrun a buffer when decrypting SSL.

Discovered by Noam Rathaus.

Change-Id: Ia0275601b2a825ba616656064d9a6eca109e34fa
Reviewed-on: https://code.wireshark.org/review/6256
Petri-Dish: Gerald Combs <gerald@wireshark.org>
Reviewed-by: Gerald Combs <gerald@wireshark.org>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Evan Huus <eapache@gmail.com>
(cherry picked from commit d3581aecda62d2a51ea7088fd46975415b03ec57)
Conflicts:
	epan/dissectors/packet-ssl-utils.c
Reviewed-on: https://code.wireshark.org/review/6325
(cherry picked from commit 7fc3a06b9bdcb4cb5760a3b55aa9bc76ff135d2c)
Reviewed-on: https://code.wireshark.org/review/6431
Reviewed-by: Michael Mann <mmann78@netscape.net>
---
 epan/dissectors/packet-ssl-utils.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/epan/dissectors/packet-ssl-utils.c b/epan/dissectors/packet-ssl-utils.c
index 4051933..9e78d18 100644
--- a/epan/dissectors/packet-ssl-utils.c
+++ b/epan/dissectors/packet-ssl-utils.c
@@ -2558,6 +2558,10 @@ ssl_decrypt_record(SslDecryptSession*ssl,SslDecoder* decoder, gint ct,
 
     /* Now strip off the padding*/
     if(decoder->cipher_suite->block!=1) {
+        if (inl < 1) { /* Should this check happen earlier? */
+            ssl_debug_printf("ssl_decrypt_record failed: input length %d too small\n", inl);
+            return -1;
+        }
         pad=out_str->data[inl-1];
         worklen-=(pad+1);
         ssl_debug_printf("ssl_decrypt_record found padding %d final len %d\n",
-- 
2.1.4

